include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/detail/error.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include "rmm/cuda_stream_view.hpp"
+
+#include <cuda_runtime_api.h>
+
+#if CUDART_VERSION >= 11020  // 11.2 introduced cudaMallocAsync
+#define CUDA_MALLOC_ASYNC_SUPPORT
```

Comment:
That's a common enough name we should scope it to prevent collision. 
```suggestion
#define RMM_CUDA_MALLOC_ASYNC_SUPPORT
```